components/airtable/new-or-modified-records-in-view.js

```diff
@@ -0,0 +1,62 @@
+const airtable = require('https://github.com/PipedreamHQ/pipedream/components/airtable/airtable.app.js')
+const moment = require('moment')
+const axios = require('axios')
+
+module.exports = {
+  name: "new-or-modified-records-in-view",
+  version: "0.0.1",
+  props: {
+    db: "$.service.db",
+    airtable,
+    baseId: { type: "$.airtable.baseId", appProp: "airtable" },
+    tableId: { type: "$.airtable.tableId", baseIdProp: "baseId" },
+    viewId: { type: "$.airtable.viewId", tableIdProp: "tableId" },
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 5,
+      },
+    },
+  },
+  async run(event) {
+    const config = {
+      url: `https://api.airtable.com/v0/${encodeURIComponent(this.baseId)}/${encodeURIComponent(this.tableId)}`,
+      params: {
+        view: this.viewId,
+      },
+      headers: {
+        Authorization: `Bearer ${this.airtable.$auth.api_key}`,
+      },
+    }
+
+    
+    const lastTimestamp = this.db.get("lastTimestamp")
+    if (lastTimestamp) {
+      config.params.filterByFormula = `LAST_MODIFIED_TIME() > "${lastTimestamp}"`
+    }
+    const timestamp = new Date().toISOString()
+    const { data } = await axios(config)
+
+    if (!data.records.length) {
+      console.log(`No new or modified records.`)
+      return
+    }
+
+    let newRecords = 0, modifiedRecords = 0
+    for (let record of data.records) {
+      if(moment(record.createdTime) > moment(lastTimestamp) || !lastTimestamp) {
```

Comment:
Reverse order of these checks! How does this make sense like this?? Plus the `!lastTimestamp` check is much faster.